pkg/runtime/runtime.go

```diff
@@ -1928,7 +1939,7 @@ func (a *DaprRuntime) publishMessageHTTP(ctx context.Context, msg *pubsubSubscri
 
 	if err != nil {
 		diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, msg.pubsub, strings.ToLower(string(pubsub.Retry)), msg.topic, elapsed)
-		return errors.Wrap(err, "error from app channel while sending pub/sub event to app")
+		return fmt.Errorf("%w from app channel while sending pub/sub event to app: %v", ErrRetriable, err)
```

Comment:
Not sure if I should handle it this way, do I need to use multierror to wrap multiple errors at the same time?

Reply:
I am not aware of multierror being used anywhere else in the code. What would be the pros and cons?

Reply:
I am referring to this [library](https://github.com/uber-go/multierr), which is currently available in the [component-contrib](https://github.com/dapr/components-contrib/search?q=multierr).
The reason I mention this is that when I use `fmt.Errorf` to wrap the lower level error as `ErrRetriable`, I can no longer do the `errors.Is` and `errors.As` check of the original lower level error for the new error. I don't know if we have such a need, but if we use multierr, we can satisfy it.

Reply:
Perhaps consider creating a custom Error class instead: https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go

Reply:
Do you mean using a custom error struct to wrap original error?

Reply:
Yes :)

Reply:
Ok, I can implement a custom error struct which has `error` field, so it can wrap original error. This custom error struct also provide `Unwrap` func, which allow user to assert the original error's type, do you think is it ok?

Reply:
Perfect :)

Reply:
Finished, please take a look :)